consensus/polybft/polybft.go

```diff
@@ -0,0 +1,662 @@
+		executor:   p.config.Executor,
+	}
+
+	// initialize pbft engine
+	opts := []pbft.ConfigOption{
+		pbft.WithLogger(p.logger.Named("Pbft").
+			StandardLogger(&hclog.StandardLoggerOptions{}),
+		),
+		pbft.WithTracer(otel.Tracer("Pbft")),
+	}
+
+	// create pbft topic
+	p.pbftTopic, err = p.config.Network.NewTopic(pbftProto, &proto.GossipMessage{})
+	if err != nil {
+		return fmt.Errorf("failed to create pbft topic. Error: %w", err)
+	}
+
+	p.pbft = pbft.New(p.key, &pbftTransportWrapper{topic: p.pbftTopic}, opts...)
+
+	// check pbft topic - listen for transport messages and relay them to pbft
+	err = p.pbftTopic.Subscribe(func(obj interface{}, from peer.ID) {
+		gossipMsg, _ := obj.(*proto.GossipMessage)
+
+		var msg *pbft.MessageReq
+		if err := json.Unmarshal(gossipMsg.Data, &msg); err != nil {
+			panic(err)
+		}
+
+		p.pbft.PushMessage(msg)
+	})
+
+	if err != nil {
+		return fmt.Errorf("topic subscription failed: %w", err)
+	}
+
+	// create bridge topic
+	bridgeTopic, err := p.config.Network.NewTopic(bridgeProto, &proto.TransportMessage{})
+	if err != nil {
+		return fmt.Errorf("failed to create bridge topic. Error: %w", err)
+	}
+	// set pbft topic, it will be check if/when the bridge is enabled
+	p.bridgeTopic = bridgeTopic
+
+	// set block time
+	p.blockTime = time.Duration(p.config.BlockTime)
+
+	// initialize polybft consensus data directory
+	p.dataDir = filepath.Join(p.config.Config.Path, "polybft")
+	// create the data dir if not exists
+	if err := os.MkdirAll(p.dataDir, 0755); err != nil {
+		return fmt.Errorf("failed to create data directory. Error: %w", err)
+	}
+
+	stt, err := newState(filepath.Join(p.dataDir, stateFileName), p.logger)
+	if err != nil {
+		return fmt.Errorf("failed to create state instance. Error: %w", err)
+	}
+
+	p.state = stt
+	p.validatorsCache = newValidatorsSnapshotCache(p.config.Logger, stt, p.consensusConfig.EpochSize, p.blockchain)
+
+	return nil
+}
+
+// Start starts the consensus and servers
+func (p *Polybft) Start() error {
+	p.logger.Info("starting polybft consensus")
+
+	// start syncer
+	if err := p.startSyncing(); err != nil {
+		return err
+	}
+
+	// start consensus
+	return p.startSealing()
+}
+
+// startSyncing starts the synchroniser
+func (p *Polybft) startSyncing() error {
+	if err := p.syncer.Start(); err != nil {
+		return fmt.Errorf("failed to start syncer. Error: %w", err)
+	}
+
+	go func() {
+		nullHandler := func(b *types.Block) bool {
+			return false
+		}
+
+		if err := p.syncer.Sync(nullHandler); err != nil {
+			panic(fmt.Errorf("failed to sync blocks. Error: %w", err))
+		}
+	}()
+
+	return nil
+}
+
+// startSealing is executed if the PolyBFT protocol is running in sealing mode.
+func (p *Polybft) startSealing() error {
+	p.logger.Info("Using signer", "address", p.key.String())
+
+	if err := p.startRuntime(); err != nil {
+		return fmt.Errorf("consensus runtime start failed: %w", err)
+	}
+
+	go func() {
+		// start the pbft process
+		p.startPbftProcess()
+	}()
```

Comment:
Why run this in a go routine?

Reply:
@Nemanja0x

Reply:
`statPbftProcess` is a blocking function. It starts infinite loop to call PBFT engine. Furthermore, it is a part of `consensus.Start()` function and `consensus.Start()` must not be a blocking function since it is called during the initialization  of server - `NewServer()`.